### PR TITLE
COMDOX-83: Migrate GraphQL Developer Guide from magento/devdocs

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Purpose of this pull request

This pull request (PR) migrates the [GraphQL Developer Guide](https://devdocs.magento.com/guides/v2.4/graphql/index.html) from the `devdocs.magento.com` site to the Adobe DevSite.

- **Staging**: https://developer-stage.adobe.com/commerce/webapi/graphql/
- **Related to**: https://github.com/magento-commerce/devdocs/pull/3215
- **Jira:** COMDOX-83

### Link checker updates

I've updated the npm script for the link checker to ignore the `src/pages/_includes` and `src/pages/graphql` directories because the link checker fails on anchors that use custom header IDs. 

These anchors seem to work on the rendered pages, but the link checker flags them as errors. I'm conflicted about ignoring these directories and would prefer to remove the custom header IDs instead. I'm still thinking about it.

> **UPDATE:** 
>I stripped out all custom header IDs because they were unnecessary and updated anchors to use the default header IDs instead. It was the only way to ensure all links were working properly after migration. I still have to ignore the graphql and includes directory in the link check npm script though because links to anchors in includes fail as false positives due to the way the tool checks links before compiling HTML. There are 14 false positives:

```src/pages/_includes/graphql/customer-orders-output.md
    63:54-63:73  warning  Link to unknown heading: `returns`                                                                               missing-heading  remark-validate-links

src/pages/_includes/graphql/customer-output.md
      3:15-3:61  warning  Link to unknown heading: `customeraddress-attributes`                                                            missing-heading  remark-validate-links

src/pages/graphql/schema/customer/mutations/create-address.md
    84:26-84:77  warning  Link to unknown heading: `customeraddressinput-attributes`                                                       missing-heading  remark-validate-links

src/pages/graphql/schema/customer/mutations/update-address.md
    55:26-55:77  warning  Link to unknown heading: `customeraddressinput-attributes`                                                       missing-heading  remark-validate-links

src/pages/graphql/schema/customer/queries/customer.md
  693:19-693:59  warning  Link to unknown heading: `rewardpoints-attributes`                                                               missing-heading  remark-validate-links

src/pages/graphql/schema/orders/interfaces/credit-memo-item.md
    30:20-30:86  warning  Link to unknown heading: `itemselectedbundleoption-attributes`                                                   missing-heading  remark-validate-links
    46:24-46:86  warning  Link to unknown heading: `downloadableitemslinks-attributes`                                                     missing-heading  remark-validate-links
    58:15-58:55  warning  Link to unknown heading: `giftcarditem-attributes`                                                               missing-heading  remark-validate-links

src/pages/graphql/schema/orders/interfaces/invoice-item.md
    30:20-30:86  warning  Link to unknown heading: `itemselectedbundleoption-attributes`                                                   missing-heading  remark-validate-links
    42:24-42:86  warning  Link to unknown heading: `downloadableitemslinks-attributes`. Did you mean `downloadableinvoiceitem-attributes`  missing-heading  remark-validate-links
    54:15-54:55  warning  Link to unknown heading: `giftcarditem-attributes`                                                               missing-heading  remark-validate-links

src/pages/graphql/schema/orders/interfaces/order-item.md
    32:20-32:86  warning  Link to unknown heading: `itemselectedbundleoption-attributes`                                                   missing-heading  remark-validate-links

src/pages/graphql/schema/orders/interfaces/shipment-item.md
    29:20-29:86  warning  Link to unknown heading: `itemselectedbundleoption-attributes`                                                   missing-heading  remark-validate-links

src/pages/graphql/schema/orders/mutations/add-return-tracking.md
    78:30-78:90  warning  Link to unknown heading: `returnshippingtracking-attributes`                                                     missing-heading  remark-validate-links

⚠ 14 warnings
```

## TODO

- [x] Fix nested includes
- [x] Address liquid template in graphql include file
- [ ] Add description metadata?
- [ ] Add text to placeholder index files?

## Checklist

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My changes follow the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jeff-matthews** [comment]
@keharper, this PR is ready for review. I deployed a fresh staging build.

I'm reluctant to add description metadata because I'm afraid it'll be a wasted effort after we add autogenerated GraphQL API docs (#56).

**Comment by jeff-matthews** [comment]
@keharper, I take it back. I'm still finding issues to fix with the TOC/navigation. I'll let you know when I'm done.

**Comment by jeff-matthews** [comment]
@keharper, PR is ready for review now. A new staging build is in progress.

File: src/pages/graphql/schema/b2b/requisition-list/mutations/delete-items.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+title: deleteRequisitionListItems mutation
+edition: b2b
+contributor_name: EY
+---
+
+# deleteRequisitionListItems mutation
+
+The `deleteRequisitionListItems` mutation removes items from the specified requisiton list for the logged in customer.
+
+This mutation requires a valid [customer authentication token](../../../customer/mutations/generate-token.md).
```
Comment (keharper) [inline src/pages/graphql/schema/b2b/requisition-list/mutations/delete-items.md:11]: This link doesn't render. I don't understand why.

**Review by keharper — COMMENTED** [review]
Almost there

File: src/pages/graphql/schema/b2b/requisition-list/mutations/delete-items.md (hunk 1 above)
Comment (jeff-matthews) [reply src/pages/graphql/schema/b2b/requisition-list/mutations/delete-items.md:11]: Weird... it render ok for me.

Screen Shot 2022-11-21 at 4 31 29 PM

Comment (jeff-matthews) [reply src/pages/graphql/schema/b2b/requisition-list/mutations/delete-items.md:11]: The error was in https://developer-stage.adobe.com/commerce/webapi/graphql/schema/b2b/requisition-list/mutations/delete/. Fixed in https://github.com/AdobeDocs/commerce-webapi/pull/57/commits/a42d9fb0d9a8a643d0e6a88e6e74129dc89a9022.

**Review by keharper — APPROVED** [review]

**Pull request merged**